### PR TITLE
Clean up Sentry.Envelope

### PR DESCRIPTION
**Pull request by whatyouhide**
* Improved encoding performance
* Reduced API surface (this is a private module, so no public-facing changes)
* Reduced amount of code

### CODE REVIEW

File: lib/sentry/client.ex [hunk 1]
```diff
@@ -72,8 +72,8 @@ defmodule Sentry.Client do
   @spec encode_and_send(Event.t(), result()) :: send_event_result()
   defp encode_and_send(event, result) do
     result =
-      Sentry.Envelope.new()
-      |> Sentry.Envelope.add_event(event)
```
Comment (whatyouhide) [inline lib/sentry/client.ex:76]: We only ever add a single event, so no need for `add_event/2`.

File: lib/sentry/envelope.ex [hunk 2]
```diff
@@ -1,145 +1,127 @@
...
-    |> Map.put(:event_id, event_id)
-    |> Map.update!(:items, fn items ->
-      items ++ [event]
-    end)
+  @spec to_binary(t()) :: {:ok, binary()} | {:error, any()}
+  def to_binary(%__MODULE__{} = envelope) do
+    json_library = Config.json_library()
+
+    encoded_items =
+      Enum.map(envelope.items, fn item ->
```
Comment (whatyouhide) [inline lib/sentry/envelope.ex:34]: `Enum.map/2` + `throw`/`catch` using iodata is really good for speed 😉

File: lib/sentry/envelope.ex [hunk 3]
```diff
@@ -1,145 +1,127 @@
...
-    envelope
-  end
-
-  # Returns the event in the envelope if one exists.
-  @spec event(t()) :: Event.t() | nil
-  def event(envelope) do
-    envelope.items
-    |> Enum.filter(fn item -> is_event?(item) end)
-    |> List.first()
-  end
```
Comment (whatyouhide) [inline lib/sentry/envelope.ex:93]: Moved these to a test helper.

**Comment by sl0thentr0py** [comment]
i would revert this tbh, we need this later

**Pull request merged**